ios/engine/KMEI/KeymanEngine/Classes/Keyboard/InputViewController.swift

```diff
@@ -333,6 +333,60 @@ open class InputViewController: UIInputViewController, KeymanWebDelegate {
       }
     }
   }
+  
+  func deleteSelection() -> Bool {
+    if let selected = textDocumentProxy.selectedText, selected.count > 0 {
+      /*
+        Since we're doing some funky text manipulation, it's best to add
+        a "canary" check in case something does go awry with it in
+        the future.
+      */
+      let beforeManipulation = textDocumentProxy.documentContextBeforeInput ?? ""
+
+      /*
+        Apple has some special nuances to its text deletion that our internal
+        Web engine does not emulate.
+       
+        .deleteBackward() behaviors:
+        - If there is selected text immediately following a space (U+0020),
+          it will delete that space IN ADDITION to the selected text.
+        - If there is selected text that starts mid-word, it will NOT delete
+          the preceding character.
+       
+        Compare to Web:  Web states an exact number of characters to delete
+        before the start of the currently-selected range... and it does this
+        completely unaware of the nuances listed above for .deleteBackward().
+        Keyman keyboard rules are likewise unaware of Apple's nuances.
```

Comment:
This is a bit of a side-track. 

```suggestion
        Our policy (#9073) on handling the backspace key when there is a 
        text selection is to just delete the selection. We have to override
        the special case of space being deleted by .deleteBackward() ourselves.
```